src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -725,7 +725,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 
     def setFlag(mask: Long): this.type   = { _rawflags |= mask ; this }
     def resetFlag(mask: Long): this.type = { _rawflags &= ~mask ; this }
-    def resetFlags() { rawflags &= TopLevelCreationFlags }
+    def resetFlags() { rawflags = 0 }
```

Comment:
It is worth considering that this code is used not only in resident compilation, but also in single-run compilation when entering a `ClassDef` that redefines a class symbol on the compilation classpath.  See `enterClassSymbol` / `updatePosFlags`.

In any case, I'm struggling to understand the motivation for `TopLevelCreationFlags`. 

The `JAVA` flag is also one that should not be persisted: if the current compilation run redefines a class on the classpath that used to be Java compiled, we should forget the `JAVA` flag.

Maybe there is something more nuanced about `PACKAGE` / `MODULE`.